src/ray/core_worker/transport/dependency_resolver.cc

```diff
@@ -71,49 +81,74 @@ void InlineDependencies(
 }
 
 void LocalDependencyResolver::ResolveDependencies(TaskSpecification &task,
-                                                  std::function<void()> on_complete) {
+                                                  std::function<void(Status)> on_complete) {
   absl::flat_hash_map<ObjectID, std::shared_ptr<RayObject>> local_dependencies;
+  std::vector<ActorID> actor_dependences;
   for (size_t i = 0; i < task.NumArgs(); i++) {
     if (task.ArgByRef(i)) {
       local_dependencies.emplace(task.ArgId(i), nullptr);
     }
+    for(const auto& in : task.ArgInlinedRefs(i)) {
```

Comment:
Is this actually capturing nested refs? For example, suppose

```
actor = Actor.remote()
nested_ref = ray.put(actor)
f.remote(nested_ref)
```

cc @stephanie-wang

Reply:
It won't capture the nested refs. Should I also add the support for this here?

Reply:
Hmm, we should if it's easy. Alternatively we could add a barrier prior to returning from any ray.put() call? I think that would also solve it, without needing the complexity of nested id tracking.

Reply:
Sorry I don't have much context here, but is this to make sure that an actor is created before passing its handle to another task? I'm not sure if this step is necessary.

Reply:
@stephanie-wang  sorry for miss your message here. This is to ensure the actor is **registered** before passing its handle to another task.